catalog/video/catalog-mpegts-stream/src/test/java/com/connexta/alliance/video/stream/mpegts/TestFileExtensionFilenameGenerator.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) Connexta, LLC
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.alliance.video.stream.mpegts;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.Assert.assertThat;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class TestFileExtensionFilenameGenerator {
+
+    private static final String EXT = "xyz";
+
+    private FileExtensionFilenameGenerator fileExtensionFilenameGenerator;
+
+    @Before
+    public void setup() {
+        fileExtensionFilenameGenerator = new FileExtensionFilenameGenerator(EXT);
+    }
+
+    @Test
+    public void testGenerateFilenameAddingExt() {
+        assertThat(fileExtensionFilenameGenerator.generateFilename("a"), is("a." + EXT));
+    }
+
+    @Test
+    public void testGenerateFilenameExtAlreadyPresent() {
+        assertThat(fileExtensionFilenameGenerator.generateFilename("a." + EXT), is("a." + EXT));
+    }
+
+    @Test
+    public void testGenerateFilenameMissingDot() {
```

Comment:
The test is confirming that the code is making sure that the filename ends with ".EXT" and not just "EXT".